### PR TITLE
add group resource

### PR DESCRIPTION
**Pull request by nonylene**
# example

```ruby
group "example_group" do
  group_name "example" # optional
  gid 4998 # optional
end
```

cf. https://github.com/itamae-kitchen/itamae/blob/master/lib/itamae/resource/group.rb

### CODE REVIEW

**Comment by nonylene** [comment]
I've fixed to use `desired` instead of `attributes`.

File: mrblib/itamae/resource_executor/group.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+module Itamae
+  module ResourceExecutor
+    class Group < Base
+      def apply(current, desired)
+        if desired.exist
+          if exist?(desired.groupname)
+            if desired.gid && desired.gid != current.gid
+              run_specinfra(:update_group_gid, desired.groupname, desired.gid)
```
Comment (k0kubun) [inline mrblib/itamae/resource_executor/group.rb]: Why didn't you call `updated!` here? https://github.com/itamae-kitchen/itamae/blob/v1.9.9/lib/itamae/resource/group.rb#L22

File: mrblib/itamae/resource_executor/group.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            end
+          else
+            options = {
+              gid: desired.gid
+            }
+
+            run_specinfra(:add_group, desired.groupname, options)
```
Comment (k0kubun) [inline mrblib/itamae/resource_executor/group.rb]: Same as https://github.com/k0kubun/itamae-mruby/pull/2/files#r78142704

**Comment by k0kubun** [comment]
I usually add `updated!` explicitly to keep compatibility without deep consideration. But for this case, I think checking by `different?` is enough. So merging this patch.

Thank you for your work!

**Pull request merged**